src/posix/platform/uart.cpp

```diff
@@ -328,55 +377,64 @@ void platformUartProcess(const fd_set *aReadFdSet, const fd_set *aWriteFdSet, co
     fd = STDOUT_FILENO;
 #endif
 
-    if ((sWriteLength > 0) && (FD_ISSET(fd, aWriteFdSet)))
+    if ((FD_ISSET(fd, aWriteFdSet)))
     {
-#if OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
-        // Don't die on SIGPIPE
-#if !defined(MSG_NOSIGNAL)
-        // some platforms (mac OS, Solaris) don't have MSG_NOSIGNAL
-        // SOME of those (mac OS, but NOT Solaris) support SO_NOSIGPIPE
-        // if we have SO_NOSIGPIPE, then set it.  otherwise, we're going
-        // to simply ignore it.
-#if defined(SO_NOSIGPIPE)
-        int err;
-        int flag;
+        otError error = platformUartWrite(fd);
 
-        flag = 1;
-        err  = setsockopt(fd, SOL_SOCKET, SO_NOSIGPIPE, &flag, sizeof(flag));
-        VerifyOrDie(err == 0, OT_EXIT_ERROR_ERRNO);
-#else
-#warning "no support for MSG_NOSIGNAL or SO_NOSIGPIPE"
-#endif
-#define MSG_NOSIGNAL 0
-#endif // !defined(MSG_NOSIGNAL)
-        rval = send(fd, sWriteBuffer, sWriteLength, MSG_NOSIGNAL);
-#else
-        rval = write(fd, sWriteBuffer, sWriteLength);
-#endif // OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
+        VerifyOrExit(error == OT_ERROR_NONE, otLogWarnPlat("UART write: %s", otThreadErrorToString(error)));
 
-        if (rval < 0)
+        if (sWriteLength == 0)
         {
+            otPlatUartSendDone();
+        }
+    }
+
+exit:
+    return;
+}
+
+otError otPlatUartFlush(void)
+{
+    otError error = OT_ERROR_NONE;
+
+    while (sWriteLength > 0)
```

Comment:
Should `sWriteLength` be updated after flush?

Reply:
it's updated in function platformUartWrite().